### ISSUE TITLE
hack not  compatible well with Chinese font

### ISSUE THREAD

**Issue by xdahu** [source-foundry/Hack#341]
Hi,guys, thanks for the works on the great font for coding.
It seems the hack font not comatile well with chinese font on eclipse. the line height is wrong. maybe is eclipse bug,but when use go mono font or consolas is all right.

on other editor, hack font seem always smaller than chinese font, but donnot have line height issue.

![2017-10-20_114734](https://user-images.githubusercontent.com/24514012/31804805-920e63ba-b58e-11e7-8751-61cda395ff4a.png)

**Comment by chrissimpkins** [comment]
@letconst thanks for the issue report.  Can you point us to the the typeface that you are using to display the CJK glyphs with Hack?   We can take a look at the metrics.

**Comment by letconst** [comment]
@chrissimpkins , the font in the attachment is Simsun font, fallbak font use which chinese font is all depend on system and software setting, the most used fallback chinese font is simsun and micro Yahei iin China Mainland.

**Comment by chrissimpkins** [comment]
Thanks.  Let me see if I can find a copy to examine.

**Comment by chrissimpkins** [comment]
As we look into the issue, you can use the following instructions to modify the line spacing and see if you can find a vertical metric that makes it appear better.  

https://github.com/source-foundry/Hack#line-spacing-adjustments

Unclear if this is a vertical metrics issue.

**Comment by chrissimpkins** [comment]
Are you willing to explore a few attempts a line spacing adjustments in Hack to see if it addresses this?

**Comment by xdahu** [comment]
@chrissimpkins,  maybe on weekend, I am inexperienced on font adjustments, so it seems a little complicate, but I will try if have time at the weekend

**Comment by chrissimpkins** [comment]
Actually very simple.  Here is a script that will execute the entire line spacing adjustment for you:

https://gist.github.com/chrissimpkins/f39e85f5f157d839e46168de1b61a174

Just change the `PERCENT_UPM` value in the script to 30, save and execute the script in the directory that contains a set of the Hack fonts.  Once they are patched with new line spacing values, reinstall and try again.

Post images with that adjustment and I would be happy to advise further based upon the result.

Thanks!

**Comment by chrissimpkins** [comment]
Sorry didn't realize you were on Windows.  If you don't have the ability to execute a shell script, you can use the following approach (Python 2 or 3 dependency for this):

```
// from inside directory that contains copies of the Hack fonts
$ pip install font-line
$ font-line font-line percent 30 Hack-Regular.ttf Hack-Bold.ttf Hack-Italic.ttf Hack-BoldItalic.ttf
```

You should receive confirmation that the patch worked and then you can remove existing Hack fonts, reinstall the patched versions, clear cache/reboot, and give it a try again.  

This is increasing the vertical metrics of the Hack fonts in the OpenType tables.  My goal is to see if we stretch the line spacing a bit you see improved alignment with the CJK fallback metrics that you are using.

**Comment by chrissimpkins** [comment]
@be5invis Belleve, do you happen to have any thoughts about this issue?  I am guessing that we are dealing with a vertical metrics difference between the Latin glyphs (from Hack) and CJK glyphs (from Simsun), but have no experience with this.

**Comment by chrissimpkins** [comment]
Will close.  Please reopen when you are available to test changes that might address this issue.  Thanks!

**Comment by Davidy22** [comment]
Found this issue while checking to see if anyone has requested correct monospacing in the hack font between chinese and other glyphs, went and "ran" the script real quick. Installing font-line didn't actually install anything to my path, I ended up cowboying the source code into the same directory as the font files and running `commands.modify_linegap_percent("Hack-Regular.ttf", "30")` from inside the python interpreter to generate a linegap30 version. The regular hack font:
![Screenshot from 2022-01-27 22-41-10](https://user-images.githubusercontent.com/872968/151381056-51686ad8-f17d-4ba5-8382-8f85e8f4636c.png)

And modified hack font:
![Screenshot from 2022-01-27 22-40-18](https://user-images.githubusercontent.com/872968/151380884-ee60f58a-991a-47c6-b1ba-1b6e55cd96a6.png)

Honestly I can't really tell a difference, height wise,  but I did notice that the chinese characters are still not monospaced properly, which was what I came here initially to address anyways. Is there another function in font-line we might try to make monospacing line up properly or is this something we need to make glyphs for the whole language for? I am not a good artist, the best I could probably do is grab existing glyphs from another open cn font and resize them to fit hack.

As a side note, I noticed coming here that CI's been red for a while, is that alright?
